core/src/test/scala/akka/kafka/scaladsl/IntegrationSpec.scala

```diff
@@ -5,113 +5,32 @@
-import org.apache.kafka.common.serialization.{ByteArrayDeserializer, ByteArraySerializer, StringDeserializer, StringSerializer}
-import org.scalactic.TypeCheckedTripleEquals
 import org.scalatest._
-import org.scalatest.concurrent.Eventually
 
 import scala.collection.JavaConverters._
 import scala.concurrent.duration._
 import scala.concurrent.{Await, Future}
-import scala.language.postfixOps
-
-class IntegrationSpec extends TestKit(ActorSystem("IntegrationSpec"))
-  with WordSpecLike with Matchers with BeforeAndAfterAll
-  with BeforeAndAfterEach with TypeCheckedTripleEquals with Eventually {
-
-  implicit val stageStoppingTimeout = StageStoppingTimeout(15.seconds)
-  implicit val mat = ActorMaterializer()(system)
-  implicit val ec = system.dispatcher
-  implicit val embeddedKafkaConfig = EmbeddedKafkaConfig(9092, 2181, Map(
-    "offsets.topic.replication.factor" -> "1",
-    "offsets.retention.minutes" -> "1",
-    "offsets.retention.check.interval.ms" -> "100"
-  ))
-  val bootstrapServers = s"localhost:${embeddedKafkaConfig.kafkaPort}"
-  val InitialMsg = "initial msg in topic, required to create the topic before any consumer subscribes to it"
-
-  override protected def beforeAll(): Unit = {
-    super.beforeAll()
-    EmbeddedKafka.start()
-  }
-
-  override def afterAll(): Unit = {
-    shutdown(system, 30.seconds)
-    EmbeddedKafka.stop()
-    super.afterAll()
-  }
-
-  def uuid = UUID.randomUUID().toString
-
-  def createTopic(number: Int) = s"topic$number-" + uuid
-
-  def createGroup(number: Int) = s"group$number-" + uuid
-
-  val partition0 = 0
 
-  val producerSettings = ProducerSettings(system, new ByteArraySerializer, new StringSerializer)
-    .withBootstrapServers(bootstrapServers)
-
-  def givenInitializedTopic(topic: String): Unit = {
-    val producer = producerSettings.createKafkaProducer()
-    producer.send(new ProducerRecord(topic, partition0, null: Array[Byte], InitialMsg))
-    producer.close(60, TimeUnit.SECONDS)
-  }
-
-  /**
-   * Produce messages to topic using specified range and return
-   * a Future so the caller can synchronize consumption.
-   */
-  def produce(topic: String, range: Range, settings: ProducerSettings[Array[Byte], String] = producerSettings): Future[Done] = {
-    val source = Source(range)
-      .map(n => {
-        val record = new ProducerRecord(topic, partition0, null: Array[Byte], n.toString)
-
-        Message(record, NotUsed)
-      })
-      .viaMat(Producer.flow(settings))(Keep.right)
-
-    source.runWith(Sink.ignore)
-  }
-
-  def createConsumerSettings(group: String): ConsumerSettings[Array[Byte], String] = {
-    ConsumerSettings(system, new ByteArrayDeserializer, new StringDeserializer)
-      .withBootstrapServers("localhost:9092")
-      .withGroupId(group)
-      .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
-      .withWakeupTimeout(10 seconds)
-      .withMaxWakeups(10)
-  }
+class IntegrationSpec extends SpecBase(kafkaPort = 9092) {
```

Comment:
It would be nice to factor port numbers to one single object so it is easier to track what ports are occupied and what are free. Will become espacially handly when writing new specs.